src/store/Stem.ts

```diff
@@ -0,0 +1,45 @@
+import { createSlice } from "@reduxjs/toolkit";
+import { Node } from "../utils/DataStuctures";
+
+interface StemState {
+  nodes: NodeState[];
+  activeNodeId: string | null;
+}
+
```

Comment:
Considering naming conventions:
Tree = the whole document
Branch = column
Leave = node

which is cute- or we could be straight forward:
Document,
Column,
Node

but either way I think Stem will get renamed